addons/stock_account/models/product.py

```diff
@@ -91,6 +91,8 @@ def write(self, vals):
         res = super(ProductTemplate, self).write(vals)
 
         for product_template, (products, description, products_orig_quantity_svl) in impacted_templates.items():
+            if not products.exists():
```

Comment:
You could have some records within `products` that still exists while others don't. In this case, you'd crash again. 
Example :
- Have a `product.attribute` with 3 values: A, B & C.
- Create a `product.template` with that attribute, with values A & B
- Update your `product.template` with:
  - Another category with a different valuation (e.g. FIFO)
  - Update on the attribute, to remove the value A and add the value C instead
 
In `products`, you'll have both the product for A which doesn't exist anymore and the product for B which is still valid and *should be* updated.